lib/pdf_generator/prince_pdf_generator.rb

```diff
@@ -32,4 +32,19 @@ def pdf_from_string(html_string, stylesheet)
     end
     AwesomeSpawn.run!(executable, options).output
   end
+
+  private
+
+  # Writes the stylesheet Sprocket::Asset to a file and returns the path
+  # Relies on the digest_path to keep the file fresh
+  #
+  def processed_stylesheet(stylesheet_path)
+    writer = AssetWriter.new(stylesheet_path, 'tmp/cache/pdf')
+
+    unless writer.file_exists?
+      writer.write
+    end
```

Comment:
Why should the caller be concerned about this?  Maybe the writer.write method can handle the file existence, so the caller doesn't have to care?  On that note, if the .write method returns the target path, then this whole thing becomes a one-liner, though I'd accept the argument that being more explicit could be clearer.  Here's what I'd expect the one-liner to look like:

```ruby
def processed_stylesheet(stylesheet_path)
  AssetWriter.new(stylesheet_path, 'tmp/cache/pdf').write
end
```